src/test/java/teammates/logic/core/NotificationsLogicTest.java

```diff
@@ -0,0 +1,103 @@
+    private NotificationAttributes n;
+    private final NotificationsDb notifDb = NotificationsDb.inst();
+    private final NotificationsLogic notifLogic = NotificationsLogic.inst();
+
+    @Override
+    protected void prepareTestData() {
+        // test data is refreshed before each test case
+    }
+
+    @BeforeMethod
+    public void refreshTestData() {
+        dataBundle = getTypicalDataBundle();
+        removeAndRestoreTypicalDataBundle();
+    }
+
+    @Test
+    public void testAll() throws Exception {
+        testGetNotification();
+        testCreateNotification();
+    }
+
+    private void testGetNotification() throws Exception {
+        ______TS("success: typical case");
+
+        n = getTypicalNotification();
+        notifDb.createEntity(n);
+
+        NotificationAttributes retrievedNotif = notifLogic.getNotification(n.getNotificationId());
+        assertEqual(n, retrievedNotif);
+
+        notifDb.deleteNotification(n.getNotificationId());
+        ______TS("failure: Null parameter");
+
+        assertThrows(AssertionError.class,
+                () -> notifLogic.getNotification(null));
+
+        ______TS("failure: notification doesn't exist");
+
+        assertNull(notifLogic.getNotification("nonexistant-notification"));
+    }
+
+    private void testCreateNotification() throws Exception {
+        ______TS("success: typical case");
+
+        n = getTypicalNotification();
+        notifLogic.createNotification(n);
+        // Equivalent to verifyInDatabase, since there is no getNotification endpoint
+        assertEquals(n, notifDb.getNotification(n.getNotificationId()));
+
+        notifDb.deleteNotification(n.getNotificationId());
+        ______TS("failure: Null parameter");
+
+        assertThrows(AssertionError.class,
+                () -> notifLogic.createNotification(null));
+
+        ______TS("failure: test create with invalid title name");
+
+        n = getTypicalNotification();
+        n.setTitle("");
+        Exception e = assertThrows(Exception.class, () -> notifLogic.createNotification(n));
+        assertEquals("The field 'notification title' is empty.", e.getMessage());
+        // Equivalent to verifyAbsentInDatabase, since there is no getNotification endpoint
+        assertNull(notifDb.getNotification(n.getNotificationId()));
+    }
+
+    private NotificationAttributes getTypicalNotification() {
+        return NotificationAttributes
+                .builder("some-notif-id")
+                .withStartTime(Instant.now())
+                .withEndTime(Instant.now().plusSeconds(100))
+                .withType(NotificationType.MAINTENANCE)
+                .withTargetUser(NotificationTargetUser.INSTRUCTOR)
+                .withTitle("Maintenance Notice")
+                .withMessage("Maintenance at 3pm today")
+                .build();
+    }
+
+    private void assertEqual(NotificationAttributes notif1, NotificationAttributes notif2) {
```

Comment:
Aren't two notificationattributes equal if id is the same? Why the need for this assertEquals

Reply:
Since I was not able to use the parent classes `verifyPresentInDatabase`, I was trying to verify that the database contained the notification requested by the user, therefore every single field should be the same. This was what I had proposed during our meeting

Reply:
I see. I thought that only applied to the logic method calls. Just add a comment explaining why the need for such a method 👍